benchmarks/communication/all_gather.py

```diff
@@ -11,46 +12,29 @@
 def timed_all_gather(input, output, args):
     if args.dist == 'torch':
         import torch.distributed as dist
+
+        all_gather_func = TorchBackend.get_all_gather_function()
     elif args.dist == 'deepspeed':
         import deepspeed.comm as dist
 
+        all_gather_func = dist.allgather_fn
+
     sync_all()
     # Warmups, establish connections, etc.
     for i in range(args.warmups):
-        # use all_gather_base if available
```

Comment:
The reason for this runtime-check is to enable users to benchmark communication ops using pure `torch.distributed` even without deepspeed installed (e.g. users of NeMo, Megatron-LM, fairseq, ColossalAI, etc). 

I'm of the opinion that we should preserve this behavior for the comms benchmarks (i.e. no `deepspeed.*` top-level imports and wrap all deepspeed- or torch.distributed-specific code in an `if args.dist == ''` check), but I think @tjruwase and/or @jeffra have the final say. Can one of you provide comment?

Reply:
Looks good to me.